### ISSUE TITLE
CurrentDirectory test failed in CI on Ubuntu

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#7484]
http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/ubuntu_release_tst_prtest/3614/console
```
05:32:47    System.IO.Tests.Directory_GetFileSystemEntries_str_str_so_alldirs.CurrentDirectory [FAIL]
05:32:47       System.IO.DirectoryNotFoundException : Could not find a part of the path '/mnt/j/workspace/dotnet_corefx/ubuntu_release_tst_prtest/bin/tests/Linux.AnyCPU.Release/System.IO.FileSystem.Tests/dnxcore50/slteu4ll.311'.
05:32:47       Stack Trace:
05:32:47             at System.IO.UnixFileSystem.FileSystemEnumerable`1.OpenDirectory(String fullPath)
05:32:47             at System.IO.UnixFileSystem.FileSystemEnumerable`1.<Enumerate>d__11.MoveNext()
05:32:47             at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source, Int32& length)
05:32:47             at System.IO.Directory.InternalGetFileDirectoryNames(String path, String userPathOriginal, String searchPattern, Boolean includeFiles, Boolean includeDirs, SearchOption searchOption)
05:32:47             at System.IO.Directory.GetFileSystemEntries(String path, String searchPattern, SearchOption searchOption)
05:32:47             at System.IO.Tests.Directory_GetFileSystemEntries_str.CurrentDirectory()
```

**Comment by ianhays** [comment]
Looks like there's a test somewhere that is creating a temp file in the current directory instead of the TestDirectory. I'd guess it's causing a race condition where we enumerate the entries in the current directory but by the time we hit the opendirectory call the entry no longer exists because it's been cleaned up already by the creator.

The issue would have arisen now but not before because I recently added AllDirectories as a SearchOption for all Directory tests, so that particular test directory is being opened and recursively searched now whereas before we just used TopDirectory and returned its info without opening it.
